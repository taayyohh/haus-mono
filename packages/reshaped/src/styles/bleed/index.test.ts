import getBleedStyles from "./index";

describe("Styles/Bleed", () => {
	test("handles positive value", () => {
		expect(getBleedStyles(4)).toMatchSnapshot();
	});

	test("handles 0 value", () => {
		expect(getBleedStyles(0)).toMatchSnapshot();
	});

	test("handles undefined value", () => {
		expect(getBleedStyles()).toMatchSnapshot();
	});

	test("handles responsive value", async () => {
		expect(getBleedStyles({ s: 4, m: 0, l: 2 })).toMatchSnapshot();
	});
});
