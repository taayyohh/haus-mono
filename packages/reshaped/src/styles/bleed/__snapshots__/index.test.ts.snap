// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Styles/Bleed > handles 0 value 1`] = `
{
  "classNames": [
    "_root_f6b186",
    "_--bleed-false_f6b186",
  ],
  "variables": {
    "--rs-bleed-s": 0,
  },
}
`;

exports[`Styles/Bleed > handles positive value 1`] = `
{
  "classNames": [
    "_root_f6b186",
    "_--bleed_f6b186",
  ],
  "variables": {
    "--rs-bleed-s": 4,
  },
}
`;

exports[`Styles/Bleed > handles responsive value 1`] = `
{
  "classNames": [
    "_root_f6b186",
    "_--bleed_f6b186",
    "_--bleed-false--m_f6b186",
    "_--bleed-true--l_f6b186",
  ],
  "variables": {
    "--rs-bleed-l": 2,
    "--rs-bleed-m": 0,
    "--rs-bleed-s": 4,
  },
}
`;

exports[`Styles/Bleed > handles undefined value 1`] = `null`;
